modules/unsupported/elasticsearch/src/main/java/org/geotools/process/elasticsearch/GeoHashGridProcess.java

```diff
@@ -113,53 +111,70 @@ public GridCoverage2D execute(
             throws ProcessException {
 
         try {
+            // setup sane defaults for aggregation definition, if missing
+            if (aggregationDefinition == null)
+                aggregationDefinition =
+                        defaultAggregation(
+                                argOutputEnv, argOutputWidth, argOutputHeight, obsFeatures);
+
             // construct and populate grid
             final GeoHashGrid geoHashGrid =
                     Strategy.valueOf(gridStrategy.toUpperCase()).createNewInstance();
             geoHashGrid.setParams(gridStrategyArgs);
             geoHashGrid.setEmptyCellValue(emptyCellValue);
             geoHashGrid.setScale(new RasterScale(scaleMin, scaleMax, useLog));
             geoHashGrid.initalize(
-                    argOutputEnv, obsFeatures, aggregationDefinition, queryDefinition, nativeOnly);
+                    argOutputEnv, obsFeatures, aggregationDefinition, queryDefinition);
             // convert to grid coverage
-            final GridCoverage2D nativeCoverage = geoHashGrid.toGridCoverage2D();
-
-            // reproject
-            final GridCoverage2D transformedCoverage =
-                    (GridCoverage2D)
-                            Operations.DEFAULT.resample(
-                                    nativeCoverage, argOutputEnv.getCoordinateReferenceSystem());
-            // upscale to approximate output resolution
-            final GridCoverage2D scaledCoverage =
-                    GridCoverageUtil.scale(
-                            transformedCoverage,
-                            argOutputWidth * argPixelsPerCell,
-                            argOutputHeight * argPixelsPerCell);
-            // crop (geohash grid envelope will always contain output bbox)
-            final GridCoverage2D croppedCoverage =
-                    GridCoverageUtil.crop(scaledCoverage, argOutputEnv);
-            return GridCoverageUtil.scale(croppedCoverage, argOutputWidth, argOutputHeight);
+            return geoHashGrid.toGridCoverage2D();
         } catch (Exception e) {
             throw new ProcessException("Error executing GeoHashGridProcess", e);
         }
     }
 
+    String defaultAggregation(
+            ReferencedEnvelope envelope,
+            Integer width,
+            Integer height,
+            SimpleFeatureCollection obsFeatures)
+            throws FactoryException, TransformException {
+        ReferencedEnvelope wgse = envelope.transform(DefaultGeographicCRS.WGS84, true);
+        double cellw = wgse.getWidth() / width;
+        double cellh = wgse.getHeight() / height;
+
+        // find out which GeoHash precision best fits the target width and height, we need
+        // something just a little bit more precise
+        int precision = 0;
+        double ghw = 0, ghh = 0;
+        for (; precision <= GeoHash.MAX_HASH_LENGTH; precision++) {
+            ghw = widthDegrees(precision);
+            ghh = GeoHash.heightDegrees(precision);
```

Comment:
Not sure why one would be statically imported, but not the other?

Reply:
Fair question... seems it's a leftover from some other change. Guess I was trying to prevent the formatter from folding a long line somewhere? Just a guess.